mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1009,10 +1009,24 @@ private[spark] object RandomForest extends Logging {
       // sort distinct values
       val valueCounts = valueCountMap.toSeq.sortBy(_._1).toArray
 
+      def weightedMean(pre: (Double, Int), cru: (Double, Int)): Double = {
+        val (preValue, preCount) = pre
+        val (curValue, curCount) = cru
+        (preValue * preCount + curValue * curCount) / (preCount + curCount)
```

Comment:
I'm probably over-thinking this, but do we have a possible overflow issue in the denominator? like if both are near `Int.MaxValue`. One could be converted `.toDouble` just to make sure